internal/pkg/database/database.go

```diff
@@ -623,7 +630,56 @@ func (db *myDB) RoadSegmentSurfaceUpdated(segmentID, surfaceType string, probabi
 		}
 	}
 
-	return fmt.Errorf("Unable to update non existing RoadSegment %s", segmentID)
+	return fmt.Errorf("unable to update non existing RoadSegment %s", segmentID)
+}
+
+func (db *myDB) CreateRoadSurfaceObserved(src *diwise.RoadSurfaceObserved) (*persistence.RoadSurfaceObserved, error) {
+
+	err := validateSurfaceType(src.SurfaceType.Value)
+	if err != nil {
+		return nil, err
+	}
+
+	if src.SurfaceType.Probability <= 0 || src.SurfaceType.Probability > 1 {
+		return nil, fmt.Errorf("probability %f is not within acceptable range: (0, 1.0]", src.SurfaceType.Probability)
+	}
+
+	lat := src.Position.Value.Coordinates[0]
```

Comment:
This a bug. Position 0 is the longitude according to https://tools.ietf.org/html/rfc7946#section-3.1.1.